pkg/prometheus/watcher.go

```diff
@@ -355,6 +411,84 @@ func (w *Watcher) createSvc(obj metav1.ObjectMeta, kind string, selector map[str
 	return annotations, nil
 }
 
+func (w *Watcher) formatAlertConfig(params []monitoringv1.Param) ([]byte, error) {
+	var url, channel string
+	for _, param := range params {
+		log.Debugf("Params: %s %s", param.Name, param.Value)
+		if param.Name == "url" {
+			url = param.Value
+		}
+		if param.Name == "channel" {
+			channel = param.Value
+		}
+	}
+
+	if url == "" {
+		log.Error("url field missing in slack config")
+		return nil, os.ErrInvalid
+	}
+
+	if channel == "" {
+		log.Error("channel field missing in slack config")
+		return nil, os.ErrInvalid
+	}
+
+	skcfg := fmt.Sprintf("  slack_configs:\n  - api_url: '%s'\n    channel: '%s'", url, channel)
```

Comment:
Please suggest if there is a more elegant way of doing this, we are trying to append a yaml to another yaml file here. I tried yaml Marshal methods but this needs to be correctly indented with the original file.

Reply:
You can represent this as an internal struct. e.g. https://github.com/platform9/sentry/blob/master/pkg/repo/structs.go.
I belive appbert does something similar to read package manifest. Then use yaml.MarshalIndent function.

Reply:
Is there a yaml.MarshalIndent function, I didn't find one in appbert, also had looked into gopkg.in/yaml.v2 earlier but did not find a way to add indentation.

Reply:
Fixed this by reading everything in a structure, appending receivers as required and then a yaml.Marshal writes out the entire config file from the structure. Partially updating a yaml file is not longer required.